runtime/src/snapshot_utils.rs

```diff
@@ -155,6 +155,49 @@ impl Ord for BankSnapshotInfo {
     }
 }
 
+impl BankSnapshotInfo {
+    pub fn new_from_dir(
+        bank_snapshots_dir: impl AsRef<Path>,
+        slot: Slot,
+    ) -> Option<BankSnapshotInfo> {
+        // check this directory to see if there is a BankSnapshotPre and/or
+        // BankSnapshotPost file
+        let bank_snapshot_dir = get_bank_snapshots_dir(&bank_snapshots_dir, slot);
+        let bank_snapshot_post_path = bank_snapshot_dir.join(get_snapshot_file_name(slot));
+        let mut bank_snapshot_pre_path = bank_snapshot_post_path.clone();
```

Comment:
minor nit/optimization:
we only ever use the post path if the pre path does not exist. If we have a good way to directly get the pre path, it'd be better to check that, and only then generate the post path. Right now we have an extra PathBuf allocation.

Feel free to leave it as is for this PR

Reply:
Thanks!  Agreed.  I will make this change in the next PR.  not worth rerun the auto tests just for this.